flow360/component/utils.py

```diff
@@ -466,6 +469,11 @@ def detect(cls, file: str):
         """
         detects compression from filename
         """
+        if CompressionFormat.TARGZ.ext() in file.lower() and file[-7:].lower().endswith(
```

Comment:
Here is to check whether the file ends with .tar.gz, which has 7 letters. Basically the os.path.splitext() doesn't work here since the extension has 2 dots, so file[-7:] is required.

Reply:
why not then just
```python
if file.lower().endswith(CompressionFormat.TARGZ.ext()):
...
```
?

Reply:
for the return you can use regexp or at least:
```python
return CompressionFormat.TARGZ, file[:-1*len(CompressionFormat.TARGZ.ext())]
```

Reply:
good point! This is a very smart solution. Fixed.